ui/src/components/Forms/Tokens/MintForm.tsx

```diff
@@ -142,6 +139,70 @@ export const MintForm: React.FC = () => {
             </FormControl>
           </Grid>
         </Grid>
+        <Grid item xs={12}>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={withMessage}
+                onChange={() => {
+                  if (withMessage) {
+                    setJsonPayload({
+                      pool: pool?.name,
+                      amount: amount.toString(),
+                      messagingMethod: null,
+                    });
+                  }
+                  setWithMessage(!withMessage);
+                }}
+              />
+            }
+            label={t('mintWithData')}
+          />
+        </Grid>
+        {withMessage === true && (
```

Comment:
This whole block should be componentized and re-used across forms

Reply:
great call